polaris-react/src/components/Button/Button.tsx

```diff
@@ -200,6 +201,16 @@ export function Button({
     setDisclosureActive((disclosureActive) => !disclosureActive);
   }, []);
 
+  const preventedInteraction = eventsToPrevent.reduce(
```

Comment:
Firstly, I want to say this is a very clever bit of code here, `reduce` can be a fairly tricky method to understand but you've done so here. However, it takes a little bit of time to understand fully what is going on here, we should try and make our code that we write as clear as possible for other folks that may need to edit it.

Furthermore, if `disabled` is true, and we spread the `preventedInteraction` variable on the `button` like in line 247 below, we are actually attaching two `onClick` handlers to the `button`. It's a better practice to include this kind of logic within the existing event handler, rather than create an alternative. So in this example, it'd look something like:

```tsx
const toggleDisclosureActive = useCallback((event: React.MouseEvent<HTMLButtonElement>) => {
  if (disabled) {
    event.preventDefault();
    return;
  }
    setDisclosureActive((disclosureActive) => !disclosureActive);
}, [disabled]);

const handleKeyPress = useCallback((event: React.KeyboardEvent<HTMLButtonElement>) => {
  if (disabled) {
    event.preventDefault();
  }
}, [disabled])
```

and where we have the button below it'd look like:

```tsx
<button
  type="button"
  className={connectedDisclosureClassName}
  aria-disabled={disabled}
  aria-label={disclosureLabel}
  aria-describedby={ariaDescribedBy}
  aria-checked={ariaChecked}
  onClick={toggleDisclosureActive}
  onMouseUp={handleMouseUpByBlurring}
  onKeyPress={handleKeyPress}
>